### PR TITLE
🌱 Add and use RepoClient API for ListStatuses

### PR DESCRIPTION
**Pull request by azeemshaikh38**
* **Please check if the PR fulfills these requirements**
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] PR title follows the guidelines defined in  https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Use a new ListStatuses for CITests check.

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No.

### CODE REVIEW

File: clients/githubrepo/statuses.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+	handler.ctx = ctx
+	handler.owner = owner
+	handler.repo = repo
+}
+
+func (handler *statusesHandler) listStatuses(ref string) ([]clients.Status, error) {
+	statuses, _, err := handler.client.Repositories.ListStatuses(handler.ctx, handler.owner, handler.repo, ref,
+		&github.ListOptions{})
+	if err != nil {
+		// nolint: wrapcheck
```
Comment (naveensrinivasan) [inline clients/githubrepo/statuses.go:44]: Can we avoid this?

**Review by naveensrinivasan — APPROVED** [review]
Thanks

File: clients/githubrepo/statuses.go (hunk 1 above)
Comment (azeemshaikh38) [reply clients/githubrepo/statuses.go:44]: I plan to remove all these linters which we seem to run `nolint` on all the time. We have too many `nolint` comments.

Comment (naveensrinivasan) [reply clients/githubrepo/statuses.go:44]: Sounds good.

**Pull request merged**